### DISCUSSION TITLE
libX11.so not found on Fedora 41 Mate

### DISCUSSION THREAD

**Discussion by gnievin** [ghostty-org/ghostty#3739] [Q&A] [answered]
Installed from copr on Fedora 41 Mate:

```
sudo dnf copr enable pgdev/ghostty
sudo dnf install ghostty 
ghostty
```
Upon starting ghostty for the first time, the following messages are displayed:

```
...
Creating EGL context version 3.2 (debug:no, forward:no, legacy:no, es:no)
Using OpenGL backend EGL
info(gtk): libadwaita version build=1.6.2 runtime=1.6.2
info(sentry): sentry envelope does not contain crash, discarding
error: FileNotFound
$ 
```

My first assumption from that error message was that it would not start without a filename to edit, but that was of course not the case.

It was not finding the libX11.so file, which was remedied with `dnf install libX11-devel`.
It now starts up successfully.

Suggest adding information to the error message about what file was not found.

**Comment by rhodes-b** [comment]
I believe this `error` text comes from GTK and not ghostty so this is not something we can change

**Reply by jcollie** [reply]
No, that's a Zig error, and I'm pretty sure it comes from here:

https://github.com/ghostty-org/ghostty/blob/6cbd69da7839260508466f9dfb2bc0c0fbb43991/src/apprt/gtk/x11.zig#L129

I'm going to work on a patch that adds a build option to specify the location of `libX11.so`.

**Comment by gnievin** [comment]
I find that it also works if I symlink libX11.so to the libX11.so.6 file provided by the installed package:
libX11-1.8.10-2.fc41.x86_64

```
$ rpm -ql libX11
/usr/lib/.build-id
/usr/lib/.build-id/b7
/usr/lib/.build-id/b7/0d757b95c23a31707a2a38bc973092186eb347
/usr/lib64/libX11.so.6
/usr/lib64/libX11.so.6.4.0
/usr/lib/.build-id
/usr/lib/.build-id/c1
/usr/lib/.build-id/c1/6cff1d9287e2e45f5b035353d84f5c4913ed0d
/usr/lib/libX11.so.6
/usr/lib/libX11.so.6.4.0
```

**Reply by rhodes-b** [reply]
Yes someone else using void linux found the same issue in the discord, it sets the wrong RPATH for x11 in some cases

**Comment by gnievin** [comment, marked as answer]
Question has been answered and more in the updated version.

The file not found is now beautifully reported in the error message thusly:

`ghostty: error while loading shared libraries: libX11.so.6: cannot open shared object file: No such file or directory
`

And in addition, it is finding the shared object in question and starts without error on Fedora 41.

Thank you!

**Reply by jcollie** [reply]
This should be fixed more permanently in 1.0.1 as we don't use `dlopen` to load libX11 anymore, it's linked as a normal dynamic library.
